src/core/algorithms/fd/fdep/fd_tree_element.cpp

```diff
@@ -243,15 +243,16 @@ void FDTreeElement::PrintDependencies(std::bitset<kMaxAttrNum>& active_path, std
     }
 }
 
-void FDTreeElement::FillFdCollection(RelationalSchema const& scheme,
-                                     std::list<FD>& fd_collection) const {
+void FDTreeElement::FillFdCollection(RelationalSchema const& scheme, std::list<FD>& fd_collection,
+                                     unsigned int max_lhs_ = -1) const {
     std::bitset<kMaxAttrNum> active_path;
-    this->TransformTreeFdCollection(active_path, fd_collection, scheme);
+    this->TransformTreeFdCollection(active_path, fd_collection, scheme, max_lhs_);
 }
 
 void FDTreeElement::TransformTreeFdCollection(std::bitset<kMaxAttrNum>& active_path,
                                               std::list<FD>& fd_collection,
-                                              RelationalSchema const& scheme) const {
+                                              RelationalSchema const& scheme,
+                                              unsigned int max_lhs_ = -1) const {
```

Comment:
also incorrect naming, also `std::numeric_limits::max()` instead of -1 with conversion (or remove the default value completely)